src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts

```diff
@@ -88,7 +90,11 @@ export const getZwaveDeviceActions = async (
   const firmwareUpdateCapabilities =
     await fetchZwaveNodeFirmwareUpdateCapabilities(hass, device.id);
 
-  if (firmwareUpdateCapabilities.firmware_upgradable) {
+  if (
+    firmwareUpdateCapabilities.firmware_upgradable &&
+    (!(await fetchZwaveIsAnyFirmwareUpdateInProgress(hass, entryId)) ||
+      (await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id)))
+  ) {
```

Comment:
Can you do this above? So its easier to read?
```suggestion
  const anyUpdatesInProgress = await fetchZwaveIsAnyFirmwareUpdateInProgress(hass, entryId);
  const deviceUpdatesInProgress = await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id);
  
  if (
    firmwareUpdateCapabilities.firmware_upgradable &&
    (!entryUpdatesInProgress || deviceUpdatesInProgress)
  ) {
```

Something like the above

Reply:
yup, I was trying to save a potential call

Reply:
You could still do that. in the var. In the second one only do the call if any updates is false.

```ts
const deviceUpdatesInProgress = anyDeviceUpdatesInProgress ? true : await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id);
```

Reply:
honestly it's probably more performant to fire all three calls and just await all of them right? (e.g. how I did it)